web/pdf_page_view.js

```diff
@@ -483,33 +454,123 @@ var PDFPageView = (function PDFPageViewClosure() {
           pageNumber: self.id,
           cssTransform: false,
         });
+      };
 
-        if (!error) {
-          resolveRenderPromise(undefined);
-        } else {
-          rejectRenderPromise(error);
+      var paintTask = this.renderer === CANVAS_RENDERER ?
+        this.paintOnCanvas(canvasWrapper) :
+        this.paintOnSvg(canvasWrapper);
```

Comment:
Moreover, it may be good to rename `canvasWrapper` to `wrapper`, or `pageWrapper` if you prefer, as it's not specific to canvases anymore.